aea/skills/base.py

```diff
@@ -427,70 +427,7 @@ def parse_module(  # pylint: disable=arguments-differ
         :param skill_context: the skill context
         :return: a list of Behaviour.
         """
-        behaviours = {}  # type: Dict[str, "Behaviour"]
-        if behaviour_configs == {}:
-            return behaviours
-        behaviour_names = set(
-            config.class_name for _, config in behaviour_configs.items()
-        )
-        behaviour_module = load_module("behaviours", Path(path))
-        classes = inspect.getmembers(behaviour_module, inspect.isclass)
-        behaviours_classes = list(
-            filter(
-                lambda x: any(
-                    re.match(behaviour, x[0]) for behaviour in behaviour_names
-                )
-                and not str.startswith(x[1].__module__, "aea.")
-                and not str.startswith(
-                    x[1].__module__,
-                    f"packages.{skill_context.skill_id.author}.skills.{skill_context.skill_id.name}",
-                ),
-                classes,
-            )
-        )
-
-        name_to_class = dict(behaviours_classes)
-        _print_warning_message_for_non_declared_skill_components(
-            skill_context,
-            set(name_to_class.keys()),
-            {
-                behaviour_config.class_name
-                for behaviour_config in behaviour_configs.values()
-            },
-            "behaviours",
-            path,
-        )
-
-        for behaviour_id, behaviour_config in behaviour_configs.items():
-            behaviour_class_name = cast(str, behaviour_config.class_name)
-            skill_context.logger.debug(
-                "Processing behaviour {}".format(behaviour_class_name)
-            )
-            if not behaviour_id.isidentifier():
-                raise AEAComponentLoadException(  # pragma: nocover
-                    f"'{behaviour_id}' is not a valid identifier."
-                )
-            behaviour_class = name_to_class.get(behaviour_class_name, None)
-            if behaviour_class is None:
-                skill_context.logger.warning(
-                    "Behaviour '{}' cannot be found.".format(behaviour_class_name)
-                )
-            else:
-                try:
-                    behaviour = behaviour_class(
-                        name=behaviour_id,
-                        configuration=behaviour_config,
-                        skill_context=skill_context,
-                        **dict(behaviour_config.args),
-                    )
-                except Exception as e:  # pylint: disable=broad-except # pragma: nocover
-                    e_str = parse_exception(e)
-                    raise AEAInstantiationException(
-                        f"An error occured during instantiation of behaviour {skill_context.skill_id}/{behaviour_config.class_name}:\n{e_str}"
-                    )
-                behaviours[behaviour_id] = behaviour
-
-        return behaviours
+        return _parse_module(path, behaviour_configs, skill_context, "behaviour")
```

Comment:
What about the "models" case? can we use the same method (with modifications if necessary) please?

Reply:
yes. The plan is to make `Model.parse_module` to behave like the others. The loading from any Python module will be handled in another helper function.